src/render/tutorials/graphql-usage/2_0.html.md

```diff
@@ -0,0 +1,777 @@
+    FragmentEventContext eventContext = new FragmentEventContext(event, clientRequest);
+
+    FragmentsHandlerOptions options = new FragmentsHandlerOptions(config);
+    ActionProvider proxyProvider = new ActionProvider(options.getAction(), supplyFactories(), vertx.getDelegate());
+    TaskBuilder taskBuilder = new TaskBuilder(FRAGMENT_TYPE, options.getTasks(), proxyProvider);
+
+    Task task = taskBuilder
+        .build(fragment)
+        .orElseThrow(() -> new NullPointerException("No task built"));
+
+    return new FragmentEventContextTaskAware(task, eventContext);
+}
+```
+
+Let's take a look at it piece by piece:
+
+```java
+JsonObject fragmentConfig = new JsonObject();
+fragmentConfig.put(FRAGMENT_TYPE, taskName);
+fragmentConfig.put("gql", new JsonObject(env.getArguments()));
+
+Fragment fragment = new Fragment(FRAGMENT_TYPE, fragmentConfig, "");
+```
+
+We create a new fragment that will be processed by tasks. It needs to know which task should start processing it, therefore `taskName` is assigned to its type. We also put graphql environment arguments to the configuration under "gql" key. Environment arguments are just values passed to parameters of operation. In our case "match" in `books(match: String)` and "id" in `book(id: String)`. By putting it into fragment configuration, we allow tasks to access it. Note that the "gql" key is arbitrary and you can call it anything you want. 
+
+For the third argument in the fragment's contructor we pass an empty string. It's our fragment's body which we're not interested in in this tutorial.
+
+```java
+RequestContext requestContext = routingContext.get(RequestContext.KEY);
+ClientRequest clientRequest = requestContext.getRequestEvent().getClientRequest();
+```
+
+We need to extract `ClientRequest` from `routingContext`. Remember the `${config.server.handlers.common.request}` part in the operation configuration in `operations.conf`? Without it there wouldn't be anything in `routingContext` under `RequestContext.KEY`.
+
+```java
+FragmentEvent event = new FragmentEvent(fragment);
+FragmentEventContext eventContext = new FragmentEventContext(event, clientRequest);
+```
+
+Our final task tree will need two things: `Task` and `FragmentEventContext`. Here we create the latter. It keeps the client request and our fragment in the form of `FragmentEvent`.
+
+```java
+FragmentsHandlerOptions options = new FragmentsHandlerOptions(config);
+ActionProvider proxyProvider = new ActionProvider(options.getAction(), supplyFactories(), vertx.getDelegate());
+TaskBuilder taskBuilder = new TaskBuilder(FRAGMENT_TYPE, options.getTasks(), proxyProvider);
+```
+
+In order to create a task we need to use `TaskBuilder`. It needs to be supplied with a few things from the configuration. We use `FragmentsHandlerOptions` to parse the configuration. We also need a supplier of action factories. We can retrieve it using standard java `ServiceLoader`. We'll take a look at `supplyFactories` method in a moment.

```

Comment:
```suggestion
In order to create a task we need to use a `TaskBuilder`. It needs to be supplied with a few things from the configuration. We use a `FragmentsHandlerOptions` to parse the configuration. We also need a supplier of action factories. We can retrieve it using the standard java `ServiceLoader`. We'll take a look at `supplyFactories` method in a moment.
```

Reply:
I was wondering about that. For example here:

>We use `FragmentsHandlerOptions` to parse the [...]

I think `FragmentsHandlerOptions` works like a proper noun and therefore don't need an article. (Just like you just say "Poland" or "Bob")

Reply:
maybe you're right, I think we use it like that everywhere else
It just sounded right to do it here, so maybe for the sake of keeping it concise, let's leave it like that

Reply:
I've adressed that ✍️